materialize/resources/resource_secret.go

```diff
@@ -122,19 +120,30 @@ func readSecretParams(id string) string {
 		WHERE mz_secrets.id = '%s';`, id)
 }
 
-//lint:ignore U1000 Ignore unused function temporarily for debugging
-type _secret struct {
-	name          sql.NullString `db:"name"`
-	schema_name   sql.NullString `db:"schema_name"`
-	database_name sql.NullString `db:"database_name"`
-}
-
 func secretRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*sqlx.DB)
 	i := d.Id()
 	q := readSecretParams(i)
 
-	readResource(conn, d, i, q, _secret{}, "secret")
+	var name, schema, database string
+	if err := conn.QueryRowx(q).Scan(&name, &schema, &database); err != nil {
```

Comment:
This has been driving me nuts. Ultimately moved away from using `StructScan` since I was having a really hard time getting the aliased columns to match up with the exposed fields. It would probably be safer to use a struct here but may have to circle back on it.